### PR TITLE
build: simple igniter configuration

### PR DESCRIPTION
**Pull request by theomessin**
## Summary of Changes

- `setup.sh` and `build.sh` scripts removed
- package.json `dependencies` all moved to `devDependencies`
- To set up run `./scripts/dev-env/run.sh npm i`
- Use igniter to build `./scripts/dev-env/run.sh npx igniter`
- `instruments`, `behavior`, and `model` steps will be skipped if no changes were made 🥳 

## Screenshots (if necessary)
image

## Testing instructions
Does the aircraft still work?

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by Benjozork** [comment]
Needs `notty` mode

**Comment by St54Kevin** [comment]
Category: QA Tester
Name: St54Kevin#3583
Date of testing: 01/02/2021
Version of the sim: 1.13.16.0
PR Tested: #3624 
Tier of testing: 1
Changes to observe: Functionality
Testing technique: 
Steps you've made to test the Pull Request:
1. Loaded up C&D
2. Went throught cockpit preparation
3. Searched for broken things

Media:

Issues (If Any): - 

Overall Rating: LGTM

Conclusions of the testing: Looks like everything works

**Comment by marcman86** [comment]
Category: QA Tester
Name: marcman86#4907
Date of testing: 25.2.2021
Version of the sim: 1.13.16.0
PR Tested: #3624
Tier of testing: 1
Changes to observe: Mod still works
Testing technique: Tryout

Steps made to test the Pull Request:

1.) Spawned C&D at EDDP
2.) Start up the bus
3.) Took off and did a small flight testing the AP

Issues: -

Overall Rating: Good

Conclusions of the testing: Seems to work

Hope this helps :-)

**Comment by ChruutvoLuzi** [comment]
Category: QA Tester
Name: ChruutvoLuzi | SD-PB.ch#8902
Date of testing: 25.2.2021
Version of the sim: 1.13.16.0
PR Tested: #3624 
Tier of testing: 1
Changes to observe: New Build Process


Testing technique: TryOut
Steps you've made to test the Pull Request:
1. Loaded C&D into LSZH 
2. Turned on the Aircraft
3. Checked all Instruments

Issues (If Any): none

Overall Rating: Good
Conclusions of the testing: Unlike Asobos World Update Building Process, this works as it should!

**Review by wpine215 — APPROVED** [review]

File: .eslintignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Math JS External Library https://mathjs.org/
 A32NX/html_ui/Pages/A32NX_Core/math.js
 build.*.js
-buildSrc/*
\ No newline at end of file
+buildSrc/*
+igniter.config.mjs
```
Comment (Benjozork) [inline .eslintignore:5]: Why is this ignored ?

**Comment by theomessin** [comment]
@Benjozork cause I couldn't get this fixed. Feel free to change this 😄 

```
/external/igniter.config.mjs
  1:1  error  Parsing error: 'import' and 'export' may appear only with 'sourceType: module'
  ```

**Comment by kberg93** [comment]
_Just gonna put in my QA anyway :)_
Category: QA Tester
Name: Kberg#0666
Date of testing: Current date 25/02/2021
Version of the sim: 1.13.16.0
PR Tested: #3624 
Tier of testing: 1
Changes to observe:
simple igniter configuration

Testing technique: Full flight.

Steps you've made to test the Pull Request: 
1. C&D LEBL, take off and flight continuing to LPPT.
2. Landed at LPPT, prepped flightplan to LXGB.
3. Departed from LPPT.
4. Landed in LXGB.



Flight notes (When done a full flight): LEBL-LPPT+ LPPT-LXGB.

Media: Insert screenshots, recordings, or anything relevant to the review here.

Issues (If Any): None noticed.

Overall Rating: Good.
Conclusions of the testing: Didnt see any issue, AP behaved good and performance was normal. Nothing broken AFAIK.

**Comment by Benjozork** [comment]
> 
> 
> @Benjozork cause I couldn't get this fixed. Feel free to change this 😄
> 
> ```
> /external/igniter.config.mjs
>   1:1  error  Parsing error: 'import' and 'export' may appear only with 'sourceType: module'
> ```

Fair enough! Not important enough atm to block this any furhter

File: scripts/build.sh [hunk 2]
```diff
@@ -1,73 +0,0 @@
-#!/bin/bash
```
Comment (Benjozork) [inline scripts/build.sh:1]: `setup` and `build` scripts need to remain so the switch to igniter is transparent to users

File: .github/workflows/pr.yml [hunk 3]
```diff
@@ -38,8 +38,8 @@ jobs:
         uses: actions/checkout@v2
       - name: Generate layout.json and update manifest.json
         run: |
-          ./scripts/dev-env/run.sh ./scripts/setup.sh
-          ./scripts/dev-env/run.sh ./scripts/build.sh
+          ./scripts/dev-env/run.sh npm install
```
Comment (Benjozork) [inline .github/workflows/pr.yml:41]: This is wrong. it needs the `--no-optional` and `--legacy-peer-deps`. See comment above.

**Comment by Benjozork** [comment]
I'll try to get on this soon

**Pull request merged**